### PR TITLE
Web onOpen added, color updates, and PSI functionality

### PR DESCRIPTION
**Pull request by theneekz**
- onOpen functionality added 
- PSI functionality added
- Colors for CrUX table made more legible. Table handles blank rows now (remain white)
- Optional appendix functionality added
- JSDocs updated

Apologies for the large PR! Will work on keeping changes small going forward.

### CODE REVIEW

File: src/sls_web.js [hunk 1]
```diff
@@ -6,12 +6,68 @@
...
+        functionName: 'filterAndSortData',
+      },
+    ];
+    spreadsheet.addMenu('Performance Starter', menuItems);
+  } catch (error) {
+    throw new Error('onOpen failed: ' + error.toString());
+  }
+}
+
+/* eslint-disable no-unused-vars*/
```
Comment (notmariazoe) [inline src/sls_web.js:37]: You won't need the no-unused-vars if you export them as above AFAIK

**Review by notmariazoe — APPROVED** [review]
LGTM!
Minor nit on linter no-unused-vars, just for the sake of consistency if nothing else.
Thank you, this is starting to look so much more decent!

File: src/sls_web.js (hunk 1 above)
Comment (theneekz) [reply src/sls_web.js]: Thank you!

**Review by notmariazoe — APPROVED** [review]
LGTM

**Pull request merged**